lib/asset/transferpart.js

```diff
@@ -118,6 +132,31 @@ class TransferPart {
     get metadata() {
         return this[PRIVATE].transferAsset.metadata;
     }
+
+    /**
+     * Creates the body that will be used for the part when transferred via HTTP.
+     * @param {PartInfo} partInfo Information about the raw part data.
+     * @returns {*} The data that should be used as the body.
+     */
+    createPartHttpBody(partInfo) {
+        // default implementation uses the data itself.
+        const { partData } = partInfo;
+        return partData;
+    }
+
+    /**
+     * Creates the headers that will be used for the part when transferred via HTTP.
+     * @param {PartHttpInfo} partHttpInfo HTTP information about the part.
+     * @returns {Headers} Simple object containing header information.
+     */
+    createPartHttpHeaders(partHttpInfo) {
+        const { httpBody } = partHttpInfo;
+        // the body could be a stream or a blob, use correct property
+        return {
+            [HTTP.HEADER.CONTENT_LENGTH]: httpBody.length || httpBody.size,
```

Comment:
Will HTTP.HEADER... always be fully set when this call happens?

Reply:
It comes from the constants file, so AFAIK it will.